hermes/probe/journal/journal.go

```diff
@@ -0,0 +1,364 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove, GitHub: evanSpendlove
+
+// Package journal implements the probe operation for checking if the Journal file
+// on the target storage system is present. It also verifies the contetns of
+// the file. The Journal stores the state of Hermes as a map of file IDs to
+// filenames. It also stores the intent to complete the next operation.
+// This allows Hermes to recover its state if it is interruped during its
+// algorithm.
+package journal
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strconv"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/go-cmp/cmp"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	"google.golang.org/api/iterator"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	cpmetrics "github.com/google/cloudprober/metrics"
+	pb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+const (
+	journalFilePrefix = "Hermes_Journal_" // Prefix of the journal filename.
+	apiLatency        = "hermes_api_latency_seconds"
+)
+
+// journalFile stores the name, checksum and contents of the Journal.
+type journalFile struct {
+	name     string
+	checksum string
```

Comment:
Not that it's wrong, but it's mildly surprising that the checksum is a string.
I would expect `[]byte`.

Reply:
It is effectively equivalent for us to store it either way, so if `[]byte` is more expected, I'll amend it.